cis_enforcements/gcp/cis_v200/section3/network.tf

```diff
@@ -0,0 +1,256 @@
+          "message": "Legacy network"
+      } -%}
+
+    {%- elif $.network.autoCreateSubnetworks != null -%}
+
+      {%- set data = {
+          "title": "Network",
+          "result": "Approved",
+          "message": "Not a legacy network"
+      } -%}
+
+    {%- else -%}
+
+      {%- set data = {
+          "title": "Subnetworks",
+          "result": "Skip",
+          "message": "No data for AutoCreateSubnetworks yet"
+      } -%}
+
+    {% endif -%}
+
+    {% set results = results.concat(data) -%}
+
+    {{ results | json }}
+  EOT
+}
+
+# GCP > Network > Firewall > Ingress Rules > Approved
+resource "turbot_policy_setting" "gcp_network_firewall_ingress_rules_approved" {
+  resource = turbot_smart_folder.main.id
+  type     = "tmod:@turbot/gcp-network#/policy/types/firewallIngressRulesApproved"
+  note     = "GCP CIS v2.0.0 - Control: 3.6 and 3.7"
+  value    = "Check: Approved"
+  # value    = "Enforce: Delete unapproved"
+}
+
+# GCP > Network > Firewall > Ingress Rules > Approved Rules
+resource "turbot_policy_setting" "gcp_network_firewall_ingress_rules_approved_rules" {
+  resource = turbot_smart_folder.main.id
+  type     = "tmod:@turbot/gcp-network#/policy/types/firewallIngressRulesApprovedRules"
+  note     = "GCP CIS v2.0.0 - Control: 3.6 and 3.7"
+  value    = <<EOT
+    REJECT $.turbot.cidr:0.0.0.0/0 $.turbot.ports=22, 3389
+
+    APPROVE *
+  EOT
+}
+
+# GCP > Network > SSL Policy > Minimum TLS Version
+resource "turbot_policy_setting" "gcp_network_ssl_policy_minimum_tls_version" {
+  resource = turbot_smart_folder.main.id
+  type     = "tmod:@turbot/gcp-network#/policy/types/sslPolicyMinimumTlsVersion"
+  note     = "GCP CIS v2.0.0 - Control: 3.9"
+  value    = "Enforce: TLS 1.2"
+}
+
+# GCP > Network > SSL Policy > Profile
+resource "turbot_policy_setting" "gcp_network_ssl_policy_profile" {
+  resource = turbot_smart_folder.main.id
+  type     = "tmod:@turbot/gcp-network#/policy/types/sslPolicyProfile"
+  note     = "GCP CIS v2.0.0 - Control: 3.9"
+  value    = "Enforce: Restricted"
+}
+
+# GCP > Network > Firewall > Approved
+resource "turbot_policy_setting" "gcp_network_firewall_approved" {
+  resource = turbot_smart_folder.main.id
+  type     = "tmod:@turbot/gcp-network#/policy/types/firewallApproved"
+  note     = "GCP CIS v2.0.0 - Control: 3.10"
+  value    = "Check: Approved"
+  # value    = "Enforce: Delete unapproved if new"
+}
+
+# GCP > Network > Firewall > Approved > Custom
+resource "turbot_policy_setting" "gcp_network_firewall_approved_custom" {
+  resource       = turbot_smart_folder.main.id
+  type           = "tmod:@turbot/gcp-network#/policy/types/firewallApprovedCustom"
+  note           = "Azure CIS v2.0.0 - Control: 3.10"
+  template_input = <<-EOT
+    {
+      firewall {
+        sourceRanges: get(path: "sourceRanges")
+        allowed: get(path: "allowed")
+      }
+    }
+  EOT
+  template       = <<-EOT
+    {%- set results = [] -%}
+
+    {%- if $.firewall.sourceRanges == null -%}
```

Comment:
I think this is risky because it could set the approvedCustom to not approved if the CMDB control is not run yet. Can remove this check I think?